contracts/StringParserLibrary.sol

```diff
@@ -46,4 +52,43 @@ library StringParserLibrary {
         }
         return NOT_EXISTING_VALUE;
     }
+
+    function countSymbolOccurrences(string memory input, string memory symbol) public pure returns (uint8) {
+        bytes memory symbolBytes = bytes(symbol);
+        require(symbolBytes.length == 1, "Method to find occurrences supports only one symbol string");
```

Comment:
You could replace the argument `string memory symbol` with something like `bytes1`. Anyway you need only one symbol to check. That would simplify the logic and saved some gas.    -1 point